packit/cli/init.py

```diff
@@ -45,12 +46,49 @@
 @click.option(
     "-f", "--force", is_flag=True, help="Reset config to default if already exists."
 )
+@click.option(
+    "--upstream-url",
+    help="URL or local path to the upstream project URL; "
+    "this implies creating a source-git repo",
+)
+@click.option(
+    "--upstream-ref",
+    help="Use this upstream git ref as a base for your source-git repo",
+)
+@click.option("--fedora-package", help="Pick spec file from this Fedora Linux package")
+@click.option(
+    "--centos-package",
+    help="Pick spec file from this CentOS Linux or CentOS Stream package",
+)
+@click.option(
+    "--dist-git-branch",
+    help="Get spec file from this downstream branch, "
+    "for Fedora this defaults to master, for CentOS it's c8s",
+)
+@click.option(
+    "--dist-git-path",
+    help="Path to dist-git repo to use. This has bigger "
+    "priority than --fedora-package and --centos-package.",
+)
+@pass_config
 @cover_packit_exception
-def init(path_or_url, force):
-    """
-    Generate new packit config.
+def init(
+    config,
+    path_or_url,
+    force,
+    upstream_url,
+    upstream_ref,
+    fedora_package,
+    centos_package,
+    dist_git_branch,
+    dist_git_path: Optional[str],
+):
     """
+    Initiate a repository to start using packit.
 
+    If you specify --upstream-url, then a source-git repository is made,
+    otherwise only configuration file packit.yaml is created.
+    """
```

Comment:
Would it be worthwhile to have a longer explanation here about what *source-git* is? Or have a *source-git* section in this man-page shortly explaining the concept, like `man packit-source-git` or something similar.

Reply:
I fear such document would be outdated in a few months. I instead added a link to our documentation, do you think that's enough?

Reply:
:+1: